### PR TITLE
DS-1730/DS-1731: Date based updates for FABS & FPDS data

### PR DESCRIPTION
**Pull request by nmonga91**
These new management commands will allow FABS and FPDS data to be updated based on the date we store which serves as the last load date. It will only take the data >= that date and run through the data to insert, delete, or update it.

### CODE REVIEW

File: usaspending_api/broker/management/commands/fabs_nightly_loader.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+        # This cascades deletes for TransactionFABS & Awards in addition to deleting TransactionNormalized records
+        TransactionNormalized.objects.\
+            filter(assistance_data__published_award_financial_assistance_id__in=ids_to_delete).delete()
+
+    @staticmethod
+    def insert_new_fabs(to_insert, total_rows):
+        logger.info('Starting insertion of new FABS data')
+
+        place_of_performance_field_map = {
```
Comment (shinson) [inline usaspending_api/broker/management/commands/fabs_nightly_loader.py]: Broker FABS now has pop_county_code and pop_country name:
column names are `place_of_perform_country_n` and `place_of_perform_country_c`

File: usaspending_api/broker/management/commands/fabs_nightly_loader.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+            "congressional_code": "place_of_performance_congr",
+            "county_name": "place_of_perform_county_na",
+            "foreign_location_description": "place_of_performance_forei",
+            "state_name": "place_of_perform_state_nam",
+            "zip4": "place_of_performance_zip4a",
+            "location_country_code": "place_of_perform_country_c"
+
+        }
+
+        legal_entity_location_field_map = {
```
Comment (shinson) [inline usaspending_api/broker/management/commands/fabs_nightly_loader.py]: Broker FABS now has: `legal_entity_city_code` and `legal_entity_country_name`

File: usaspending_api/broker/management/commands/fabs_nightly_loader.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+            nargs='+',
+            type=str,
+            help="(OPTIONAL) Date from which to start the nightly loader. Expected format: MM/DD/YYYY"
+        )
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        logger.info('Starting historical data load...')
+
+        db_cursor = connections['data_broker'].cursor()
```
Comment (tony-sappe) [inline usaspending_api/broker/management/commands/fabs_nightly_loader.py]: Curious to know the reasoning for creating the connection cursor here, only to be used in `get_fabs_data`

File: usaspending_api/broker/management/commands/fabs_nightly_loader.py (hunk 1 above)
Comment (nmonga91) [reply usaspending_api/broker/management/commands/fabs_nightly_loader.py]: Will add

File: usaspending_api/broker/management/commands/fabs_nightly_loader.py (hunk 2 above)
Comment (nmonga91) [reply usaspending_api/broker/management/commands/fabs_nightly_loader.py]: Will add

File: usaspending_api/broker/management/commands/fabs_nightly_loader.py (hunk 3 above)
Comment (nmonga91) [reply usaspending_api/broker/management/commands/fabs_nightly_loader.py]: that can be moved into `get_fabs_data`. This is likely a result from being copied from previous efforts where the cursor was used more than it is currently.

**Pull request merged**